packages/ssi-types/src/types/sd-jwt-vc.ts

```diff
@@ -325,13 +310,29 @@ export const sdJwtDecodedCredentialToUniformCredential = (
     throw Error(`JWT issuance date is required but was not present`)
   }
 
+  // Filter out the fields we don't want in credentialSubject
+  const excludedFields = new Set(['vct', 'cnf', 'iss', 'iat', 'exp', 'nbf', 'jti', 'sub'])
+  const credentialSubject = Object.entries(decodedPayload).reduce((acc, [key, value]) => {
+    if (
+      !excludedFields.has(key) &&
+      value !== undefined &&
+      value !== '' &&
+      !(typeof value === 'object' && value !== null && Object.keys(value).length === 0)
+    ) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as Record<string, any>)
+
+  // Add id to credentialSubject if applicable
+  if (sub || jti) {
+    credentialSubject.id = sub ?? jti
```

Comment:
What if id was already set by the reducer?

Reply:
fixed